mlir/lib/Dialect/Transform/IR/TransformOps.cpp

```diff
@@ -1030,6 +1041,16 @@ transform::ForeachMatchOp::apply(transform::TransformRewriter &rewriter,
 
   DiagnosedSilenceableFailure overallDiag =
       DiagnosedSilenceableFailure::success();
+
+  SmallVector<SmallVector<MappedValue>> matchInputMapping;
+  SmallVector<SmallVector<MappedValue>> matchOutputMapping;
+  SmallVector<SmallVector<MappedValue>> actionResultMapping;
+  matchInputMapping.emplace_back();
```

Comment:
do we need this outide of the call to `prepareValueMappings` given that `prepareValueMappings` immediately resizes?